docs/src/main/docs/cdi-extensions.adoc

```diff
@@ -0,0 +1,221 @@
+
+* <<Overview>>
+* <<Installing a CDI extension>>
+* <<Injecting an object using a CDI extension>>
+
+== Overview
+
+Helidon provides https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#spi[portable extensions] that you can use to inject the following objects into your applications:
+
+* HikariCP data sources
+* Jedis clients
+* Oracle Cloud Infrastructure Object Storage clients
+
+[OPEN] Add a line or two describing each service.
+
+[OPEN] We might want to explain why we picked these 3 services. Something more than just saying they're popular. And possibly mention that we’ll provide extensions for other services in the future, without naming any specifically of course.
+
+== Installing a CDI extension
+
+The Helidon CDI extensions are available as https://mvnrepository.com/artifact/io.helidon.integrations.cdi[Maven artifacts].
+
+To install an extension, ensure that the extension and its runtime dependencies are specified on your application's runtime classpath.
+
+The following examples show how to specify the required dependencies for the Helidon CDI extensions in your Maven POM (`pom.xml`)
+ 
+* To install the extension for HikariCP data sources, specify the following dependency:
+
+[source,xml]
+----
+<dependency>
+  <groupId>io.helidon.integrations.cdi</groupId>
+  <artifactId>helidon-integrations-cdi-datasource-hikaricp</artifactId>
+  <version>1.0.0</version>
+  <scope>runtime</scope>
+</dependency>
+----
+  
+* To install the extension for Jedis clients, specify the following dependency:
+ 
+[source,xml]
+----
+<dependency>
+  <groupId>io.helidon.integrations.cdi</groupId>
+  <artifactId>helidon-integrations-cdi-jedis</artifactId>
+  <version>1.0.0</version>
+  <scope>runtime</scope>
+</dependency>
+----
+  
+* To install the extension for Oracle Cloud Infrastructure Object Storage clients:
+
+   1. Clone the Oracle Cloud Infrastructure Java SDK artifacts.
+   
+      [source,bash]
+      ----
+      git clone --depth 1 --branch v1.2.44 https://github.com/oracle/oci-java-sdk.git
+      ----
+      
+   2. Install the Oracle Cloud Infrastructure Java SDK artifacts in your local Maven repository.
+   
+      [source,bash]
+      ----
+      mvn -B -U -f oci-java-sdk.git/pom.xml \
+          -Dmaven.test.skip=true \
+          -Dmaven.source.skip=true \
+          -Dmaven.javadoc.skip=true \
+          -Dlombok.delombok.skip=true \
+          -pl bmc-objectstorage \
+          -am \
+          install
+      ----  
+   
+   3. Specify the following dependency in your application's Maven POM:
+   
+      [source,xml]
+      ----
+      <dependency>
+         <groupId>io.helidon.integrations.cdi</groupId>
+         <artifactId>helidon-integrations-cdi-oci-objectstorage</artifactId>
+         <version>1.0.0</version>
+         <scope>runtime</scope>
+      </dependency>
+      ----
+
+== Injecting an object using a CDI extension
+
+The following are examples of field-injection and constructor-injection for each of the Helicon CDI extensions:
```

Comment:
Nit: "field-injection", "constructor-injection": no need for hyphens

Reply:
Fixed